### DISCUSSION TITLE
How to detect a failure on command line

### DISCUSSION THREAD

**Discussion by tahirmt** [fastlane/fastlane#17745] [Q&A] [answered]
Hi team, I want to detect a failure and report the failure from command line to slack. I'm not sure what output I would be looking for and how to access it.

**Comment by rogerluan** [comment, marked as answer]
Hey @tahirmt 👋 

Can you be more specific and maybe give examples? Which tool are you talking about?

**Reply by tahirmt** [reply]
@rogerluan Thank you for your reply. I am using `download_dsyms` with `upload_symbols_to_crashlytics` and want to be able to parse the errors if any are reported and send a notification on slack. I have this set up as a cron task in my CI so want to be notified of failures instead of having to check manually.

**Reply by rogerluan** [reply]
I believe you can add this to your ruby `Fastfile`:

```ruby
error do |lane, exception|
  slack(channel: "testing", message: exception.to_s, success: false)
end
```

(at the file top level)

**Reply by tahirmt** [reply]
Thank you @rogerluan I can't believe I missed it in the docs.

**Reply by tahirmt** [reply]
@rogerluan Do we also have a way to get the summary string in a variable that gets shown in the console? In case of success for instance.

**Reply by rogerluan** [reply]
What do you mean by summary string? I guess you could use `tee` directly when calling fastlane, but that'd be on a per-command basis, instead of per-lane. Not sure what other ways to achieve that (also unclear about your request)

**Reply by tahirmt** [reply]
Using tee could work. I just meant the result summary we get after running lanes. faltlane summary with time taken for each lane.
